runtime/convertValues.go

```diff
@@ -44,6 +44,108 @@ func ConvertEvent(event Event) cadence.Event {
+		return Value{Value: interpreter.NewAddressValueFromBytes(v.Bytes())}
+	case cadence.Int:
+		return Value{Value: interpreter.NewIntValueFromBigInt(v.Big())}
+	case cadence.Int8:
+		return Value{Value: interpreter.Int8Value(v)}
+	case cadence.Int16:
+		return Value{Value: interpreter.Int16Value(v)}
+	case cadence.Int32:
+		return Value{Value: interpreter.Int32Value(v)}
+	case cadence.Int64:
+		return Value{Value: interpreter.Int64Value(v)}
+	case cadence.Int128:
+		return Value{Value: interpreter.NewInt128ValueFromBigInt(v.Big())}
+	case cadence.Int256:
+		return Value{Value: interpreter.NewInt256ValueFromBigInt(v.Big())}
+	case cadence.UInt:
+		return Value{Value: interpreter.NewUIntValueFromBigInt(v.Big())}
+	case cadence.UInt8:
+		return Value{Value: interpreter.UInt8Value(v)}
+	case cadence.UInt16:
+		return Value{Value: interpreter.UInt16Value(v)}
+	case cadence.UInt32:
+		return Value{Value: interpreter.UInt32Value(v)}
+	case cadence.UInt64:
+		return Value{Value: interpreter.UInt64Value(v)}
+	case cadence.UInt128:
+		return Value{Value: interpreter.NewUInt128ValueFromBigInt(v.Big())}
+	case cadence.UInt256:
+		return Value{Value: interpreter.NewUInt256ValueFromBigInt(v.Big())}
+	case cadence.Word8:
+		return Value{Value: interpreter.Word8Value(v)}
+	case cadence.Word16:
+		return Value{Value: interpreter.Word16Value(v)}
+	case cadence.Word32:
+		return Value{Value: interpreter.Word32Value(v)}
+	case cadence.Word64:
+		return Value{Value: interpreter.Word64Value(v)}
+	case cadence.Fix64:
+		return Value{Value: interpreter.Fix64Value(v)}
+	case cadence.UFix64:
+		return Value{Value: interpreter.UFix64Value(v)}
+	case cadence.Array:
+		values := make([]interpreter.Value, len(v.Values))
+
+		for i, elem := range v.Values {
+			values[i] = ToRuntimeValue(elem).Value
+		}
+
+		return Value{Value: interpreter.NewArrayValueUnownedNonCopying(values...)}
+	case cadence.Dictionary:
+		keysAndValues := make([]interpreter.Value, len(v.Pairs)*2)
+
+		for i, pair := range v.Pairs {
+			keysAndValues[i*2] = ToRuntimeValue(pair.Key).Value
+			keysAndValues[i*2+1] = ToRuntimeValue(pair.Value).Value
+		}
+
+		return Value{Value: interpreter.NewDictionaryValueUnownedNonCopying(keysAndValues...)}
+	case cadence.Struct:
+		return compositeToRuntimeValue(v.StructType.Fields, v.Fields)
+	case cadence.Resource:
+		return compositeToRuntimeValue(v.ResourceType.Fields, v.Fields)
+	case cadence.Event:
+		return compositeToRuntimeValue(v.EventType.Fields, v.Fields)
+	}
+
+	panic(fmt.Sprintf("cannot convert value of type %T", value))
+}
+
+func compositeToRuntimeValue(fieldTypes []cadence.Field, fieldValues []cadence.Value) Value {
+	fields := make(map[string]interpreter.Value, len(fieldTypes))
+
+	for i := 0; i < len(fieldTypes) && i < len(fieldValues); i++ {
+		fieldType := fieldTypes[i]
+		fieldValue := fieldValues[i]
+		fields[fieldType.Identifier] = ToRuntimeValue(fieldValue).Value
+	}
+
+	return Value{Value: &interpreter.CompositeValue{Fields: fields}}
```

Comment:
Just the fields are unfortunately not enough, `Location`, `TypeID`, and `Kind` are needed as well: This information is used e.g. to link in the composite value's functions

Reply:
Yeah, I have a proposed solution for this in a follow-up PR: https://github.com/onflow/cadence/pull/45/files#diff-4893a68aef7c913546919a6adf6c0885R142-R144

Let me know what you think!

Reply:
Ah, I see! Sorry, I hadn't seen it yet 👍